inc/templates.php

```diff
@@ -25,6 +25,10 @@
  * @return array A hydrated data object.
  */
 function load_template( array $data, WP_Query $query, string $context ): array {
+
+	// Manage helmet when using template functionality.
+	add_filter( 'wp_irving_components_route', __NAMESPACE__ . '\\setup_helmet', 11, 5 );
```

Comment:
If we're always setting up header info during template loading, we should just call `setup_helmet()` directly when needed, rather than adding another filter hook during a callback, which makes testing more difficult.

Reply:
I was thinking other sites may want to enable this, and/or some sites may want to disable it?

Reply:
Yeah, that's totally reasonable. For functionality that we use by default, but think might need to be disabled, I prefer a pattern where we call it directly during our template loading but add a filter within that function to disable it if not needed instead. So something like:

```
setup_helmet( $data ) {

	$enabled = apply_filters( 'wp_irving_setup_helmet', true, $data );

	// Bail early if disabled.
	if ( ! $enabled ) {
	  return $data
	}

	// helmet logic...
}
```

This has two benefits, it makes the execution flow more predictable (we're not relying on the callback order of hooks) and makes the inclusion of helmet in the template logic more testable because our functions aren't producing side effects.